### PR TITLE
Fix TestPodSidecarLabelChangeTriggersAddIptablesAction flake

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controller/bypass/bypass_test.go [hunk 1]
```diff
@@ -25,24 +25,39 @@ import (
...
+			Name: namespaceName,
+			Labels: map[string]string{
+				"istio-injection": "enabled",
+			},
+		},
+	}
+	client := fake.NewSimpleClientset(namespace)
 	c := NewByPassController(client)
 	go c.Run(stopCh)
+	cache.WaitForCacheSync(stopCh, c.pod.HasSynced)
```
Comment (hzxuzhonghu) [inline pkg/controller/bypass/bypass_test.go:59]: This is the root cause

Comment (Okabe-Rintarou-0) [reply pkg/controller/bypass/bypass_test.go:59]: you can directly call `c.Run(stopCh)`, just remove `go`

**Comment by Okabe-Rintarou-0** [comment]
i also found that the NODE_NAME env is not aligned with the one in the test pod's `spec.nodeName`.
One is `test_node`, another is `test-node`, will it cause some problem?

**Comment by hzxuzhonghu** [comment]
They should match. Weired, why the test passed

File: pkg/controller/bypass/bypass_test.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/controller/bypass/bypass_test.go:59]: Yes,  here we can

**Comment by hzxuzhonghu** [comment]
Oh, it is because fake client does not support field selector, which actually works in kube-apiserver

**Comment by hzxuzhonghu** [comment]
ping @supercharge-xsy

**Comment by YaoZengzeng** [comment]
@hzxuzhonghu maybe this PR could be merged?

**Comment by hzxuzhonghu** [comment]
@nlgwcy I thought it was merged

**Pull request merged**